crypto/ecdsa.go

```diff
@@ -46,9 +46,10 @@ func (k *ecdsaKey) generateKey(sequence uint32) *btcec.PrivateKey {
 	copy(seed, k.PubKey().SerializeCompressed())
 	binary.BigEndian.PutUint32(seed[btcec.PubKeyBytesLenCompressed:], sequence)
 	key := newKey(seed)
-	key.D.Add(key.D, k.D).Mod(key.D, order)
-	key.X, key.Y = key.ScalarBaseMult(key.D.Bytes())
-	return key
+	key.Key = *key.Key.Add(&k.Key)
```

Comment:
Is this not missing a Modulo?

Reply:
As far as I can see, the modulo now happens inside the `Add` function, the function (from [github.com/decred/dcrec](https://github.com/decred/dcrd/blob/master/dcrec/secp256k1/modnscalar.go#L487)) is documented as
```
// Add adds the passed scalar to the existing one modulo the group order in
// constant time and stores the result in s.
```

For context: `btcec.PrivateKey` is now a `secp.PrivateKey` (containing just a `ModNScalar`) instead of an `ecdsa.PrivateKey`, so updating `X` and `Y` is no longer necessary/possible. `X`/`Y` will probably be derived later, when they are required.

(That being said, I didn't test myself, but this is covered by the test vectors in your `key_test.go`)